util/reggen/fpv_csr.sv.tpl

```diff
@@ -35,9 +35,21 @@ module ${mod_base}_csr_assert_fpv import tlul_pkg::*;
   input tl_h2d_t h2d,
   input tl_d2h_t d2h
 );
+<%
+  addr_width = rb.get_addr_width()
+  addr_msb  = addr_width - 1
+  hro_regs_list = [r for r in rb.flat_regs if not r.hwaccess.allows_write()]
+  hro_map = {}
+  hro_idx = 0
+  num_hro_regs = len(hro_regs_list)
+%>\
 
-`ifndef VERILATOR
-`ifndef SYNTHESIS
+% if num_hro_regs == 0:
+  // Currently FPV csr assertion only support HRO registers.
+% else:
+  `ifndef VERILATOR
+  `ifndef SYNTHESIS
```

Comment:
Nit: I think these lines would be better without the indentation, and similarly for the matching `endif` lines. I checked, and we've even got a [styleguide recommendation](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md#preprocessor-directives) about it. (Which I didn't expect!)